python/grass/jupyter/seriesmap.py

```diff
@@ -0,0 +1,369 @@
+            # Render image
+            img = Map(
+                width=self._width,
+                height=self._height,
+                filename=filename,
+                use_region=True,
+                env=self._env,
+                read_file=True,
+            )
+            self._render_layer(img, self._calls[i])
+        self._layers_rendered = True
+
+    def show(self, slider_width=None):
+        """Create interactive timeline slider.
+
+        param str slider_width: width of datetime selection slider
+
+        The slider_width parameter sets the width of the slider in the output cell.
+        It should be formatted as a percentage (%) between 0 and 100 of the cell width
+        or in pixels (px). Values should be formatted as strings and include the "%"
+        or "px" suffix. For example, slider_width="80%" or slider_width="500px".
+        slider_width is passed to ipywidgets in ipywidgets.Layout(width=slider_width).
+        """
+        # Lazy Imports
+        import ipywidgets as widgets  # pylint: disable=import-outside-toplevel
+
+        # Render images if they have not been already
+        if not self._layers_rendered:
+            self.render()
+
+        # Set default slider width
+        if not slider_width:
+            slider_width = "70%"
+
+        # Create lookup table for slider
+        lookup = list(zip(self._names, range(self._series_length)))
+
+        # Datetime selection slider
+        slider = widgets.SelectionSlider(
+            options=lookup,
+            value=0,
+            disabled=False,
+            continuous_update=True,
+            orientation="horizontal",
+            readout=True,
+            layout=widgets.Layout(width=slider_width),
+        )
+        play = widgets.Play(
+            interval=500,
+            value=0,
+            min=0,
+            max=self._series_length - 1,
+            step=1,
+            description="Press play",
+            disabled=False,
+        )
+        out_img = widgets.Image(value=b"", format="png")
+
+        def change_slider(change):
+            slider.value = slider.options[change.new][1]
+
+        play.observe(change_slider, names="value")
+
+        # Display image associated with datetime
+        def change_image(index):
+            # Look up layer name for date
+            filename = self._layer_filename_dict[index]
+            with open(filename, "rb") as rfile:
+                out_img.value = rfile.read()
+
+        # Return interact widget with image and slider
+        widgets.interactive_output(change_image, {"index": slider})
+        layout = widgets.Layout(
+            width="100%", display="inline-flex", flex_flow="row wrap"
+        )
+        return widgets.HBox([play, slider, out_img], layout=layout)
+
+    def save(self, filename, duration=500):
+        """
+        Creates a GIF animation of rendered layers.
+
+        Text color must be in a format accepted by PIL ImageColor module. For supported
```

Comment:
You don't have the text label implemented here. It could work the same as in TimeSeriesMap, the label would be self._names.

Reply:
I added this. It looks a bit different then the `TimeSeriesMap` implementation because that one has a bug. The default font was only available on Ubuntu so it would raise an error on Windows/Mac. 

In `SeriesMap`, if there's no font provided, it tries to set a font based on the OS. This is a better option just just using `PIL.ImageFont.load_default()` because this function doesn't take font size as an input (see [this](https://github.com/python-pillow/Pillow/issues/2695) issue).

Reply:
> I added this. It looks a bit different then the `TimeSeriesMap` implementation because that one has a bug. The default font was only available on Ubuntu so it would raise an error on Windows/Mac.

Is the bug reported?

Reply:
I fixed the bug in TimeSeriesMap in my latest commit by moving the GIF rendering function to utils.py so it can be shared by TimeSeriesMap and SeriesMap. This also reduces duplicate code which is what's currently causing the tests to fail (pylint).